### PR TITLE
octopus: Add NFFT variant

### PR DESCRIPTION
**Pull request by iamashwin99**
NFFT is an optional dependency of octopus.
This MR adds the right version of NFFT as a variant to the octopus spack package.
A variant of this MR has been tested at https://github.com/fangohr/octopus-in-spack/pull/74 to check that octopus does pick up the NFFT variant.

### CODE REVIEW

**Review by fangohr — APPROVED** [review]
This is good and should be merged. There is only one known working version of nfft at the moment, that's why the dependency is tide to that. If more versions of nfft are discovered that work, this can be extended of course.

**Review by bernhardkaindl — APPROVED** [review]
Approving after approval from @fangohr, review and build on Ubuntu 20.04 with it's gcc-10.3.0

**Pull request merged**
